lightning/src/ln/channelmanager.rs

```diff
@@ -1733,12 +1733,15 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 									}
 									if total_value >= msgs::MAX_VALUE_MSAT || total_value > data.total_msat  {
 										for htlc in htlcs.iter() {
+											let mut htlc_msat_height_data = byte_utils::be64_to_array(htlc.value).to_vec();
+											let mut height_data = byte_utils::be32_to_array(self.latest_block_height.load(Ordering::Acquire) as u32).to_vec();
+											htlc_msat_height_data.append(&mut height_data);
```

Comment:
```suggestion
											htlc_msat_height_data.extend_from_slice(&byte_utils::be32_to_array(self.latest_block_height.load(Ordering::Acquire) as u32));
```

No reason for the indirection (plus extra malloc) - 8 bytes on the stack is gonna be way faster than hitting heap.